docs/DeveloperGuide.md

```diff
@@ -2,37 +2,254 @@
+reads in any past medicine dispensing records of the patient stored in the data file. `readFile()` throws
+`CorruptedDataException` if any corruption in the data file is found in the process of reading from it, which is 
+facilitated by the `endOfFile()` method.
+
+During use of Dr Duke, if the patient decides to reset his/her diagnosis history, `resetDiagnosisHistory()` will be
+called, which will call `saveData()` after resetting the patient's diagnosis history. After the patient has been
+diagnosed with possible illnesses and dispensed appropriate medication, these records will also be saved by called
+`saveData()`. Finally, before exiting Dr Duke, `saveData()` will also be called to write all existing data to the data
+file.
+
+### Implementation of loadData() in Storage Class
+![loadDataSequenceDiagram.png](diagrams%2FloadDataSequenceDiagram.png)
+
+### Implementation of saveData() in Storage Class
+![saveDataSequenceDiagram.png](diagrams%2FsaveDataSequenceDiagram.png)
+
+[comment]: <> (//@@Geeeetyx)
+
+### Patient
+![PatientClassDiagram.png](diagrams%2FPatientClassDiagram.png)
+
+The Patient class is responsible for creating a patient object, which is required to store a user's information.
+
+The method for creating a new Patient object is as follows:
+
+~~~
+new Patient(name, hash, diagnosisHistory, medicineHistory);
+~~~
+
+Once key information about the patient is obtained from the Menu class, the new patient object is created via:
+
+~~~
+Information.storePatientInfo(hash, new Patient(name, hash, diagnosisHistory, medicineHistory));
+~~~
+
+which is found in the Menu class.
+
+#### Patient Class Methods:
+
+The methods contained in the Patient Class are primarily used in the Information, Parser and Storage class.
+
+The Patient class contains important getters and setters, allowing different classes to access the patient object and
+the patient's information via the following methods:
+
+`getName()` returns the name of the patient.
+
+`getPassword()` returns the password of the patient's account.
+
+`getPatientDiagnosisHistory()` allows for access to the array where past diagnoses are stored.
+
+`getPatientMedicineHistory()` allows for access to the array where past prescribed medications are stored.
+
+`setName()` and `setPassword()` allow the programme to set the patient's name and password.
+
+`setPatientDiagnosisHistory()` and `setPatientMedicineHistory()` allow the programme to read data from a storage file,
+and then set the stored data into the Patient object.
+
+`updatePatientMedicineHistory()` and `updatePatientDiagnosisHistory()` allow us to add new diagnoses and medications
+prescribed to the Patient object.
+
+`appendMedicineToSameDate()` appends medicine patient is prescribed if additional medicine is prescribed on the same
+day, allowing us to keep track of multiple prescriptions to one patient in one session.
```

Comment:
Should there be an explanation for self-explanatory methods such as getters and setters?